File: Year 2/Semester 4/MPP/SeventhWeekCSharpServerClient/persistence/interfaces/IRepository.cs
using SeventhWeekCSharpServerClient.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SeventhWeekCSharpServerClient.repository.interfaces
{
    public interface IRepository<ID, E> where E : Entity<ID>
    {
        List<E> GetAll();
        E FindById(ID id);
        bool Add(E entity);
        bool Delete(ID id);
        bool Update(E entity);
    }
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 11/REST/RESTJava/Persistance/src/main/java/music/rest/persistance/repository/IRepository.java
package music.rest.persistance.repository;

import rest.domain.Entity;

import java.io.Serializable;

public interface IRepository<ID extends Serializable, E extends Entity<ID>>  {
    /**
     *
     * @param id of type ID (template)
     * @return found entity or null
     */
    E findById(ID id);
    Iterable<E> getAll();
    boolean add(E entity);
    boolean delete(ID id);
    boolean update(E entity);
}
